client/EditContributionView.js

```diff
@@ -215,6 +301,42 @@ class EditContributionView extends Component {
                             href={"/preview/"+this.props.selectedContribution.name.toLowerCase().replace(/ /g, "-")}>
                                 Preview 
                     </Button>
+                    <Paper className={classes.approvalPaper} elevation={3} square={false} color={"primary"}>
+                        <FormControlLabel
+                            color={"primary"}
+                            label={approvalText}
+                            labelPlacement="bottom"
+                            control={
+                                <Switch
+                                    checked={(contrib && (contrib.approval === 'pending')) || this.state.approvalSwitch}
+                                    onChange={this.handleSwitchChange}
+                                    name="approvalSwitch"
+                                    color="secondary"
+                                /> }
+                        />
+                    </Paper>
+                    <Paper className={classes.publishPaper} elevation={3} square={false} color={"primary"}>
+                        <FormControlLabel
+                            color={"primary"}
+                            label={publishedText}
+                            labelPlacement="bottom"
+                            control={
+                                this.props.admin ? 
+                                    <Switch
+                                        name="publishedSwitch"
+                                        checked={(this.state.publishedList && (this.state.publishedList[contrib.ref.id] === 'true')) || this.state.publishedSwitch}
```

Comment:
=== true is redundant

You can prob remove publishedSwitch from state entirely and just use data from props[publishedList] once aforementioned changes to db call structure are made

Reply:
the comparison is a string comparison here as well;
You were right about the switches!